### ISSUE TITLE
[Question] RNNModel (DeepAR) clarification questions

### ISSUE THREAD

**Issue by yigitcancomlek** [unit8co/darts#1918]
Hello Developers of Darts!

Thanks once again for building such a nice package, helping us make time series predictions easier. I have two quick questions about the RNNModel (DeepAR variation). 

1) Is there a rule of thumb for selecting the beta parameter for the `GaussianLikelihood()` loss or should it also be treated as an hyperparameter to the model? Intuitively, it won't make sense to me if it is an hyperparameter as we cannot compare different models together but I would love to hear your opinions.

2) Would one able to explain the parameter `training_length`? I am unable to understand how or why it influences my prediction length, especially when I have a validation set? For instance, I have a training/validation/test time series with lengths 200,20,100 timesteps respectively. If I want to use the validation time series during fit() to implement early stopping during training, I have to limit my `training_length `to be less than 20, which results in poor predictions for the testing set. However, if I don't use the validation set and set `training_length `to be 190 (or anything higher than validation length), my predictions on the validation and test (both validation and test sets become the test set now) sets are much better. However, I need a way to stop training the model to prevent it from overfitting as if I let it run for few more epochs, it starts overfitting... So I am confused about why can't I use most (or even all) of my past training data to make predictions on the validation set, why I am confined with the validation length during the` fit()`. Isn't this also a constraint for back testing? If I wanted implement an expanding window strategy, I cannot directly use all of my past data to learn due to the `training_length` limitation. Please let me know if I am missing something about the RNNModel strategy.

Thank you very much in advance for your responses!

**Comment by madtoinou** [comment]
Hi @yigitcancomlek,

1. The original article ([link](https://arxiv.org/abs/2203.09168)) seems to recommend `beta=0.5` but in my opinion, you could consider `beta` as an hyperparameter and then compare models performance using the mean prediction ("Lβ−NLL is not meant for performance evaluation, rather it is designed to result
in meaningful gradients. Due to the weighting term, the loss value does not reflect the model’s quality.
The model performance during training should be monitored with the original negative log-likelihood
objective and, optionally, with RMSE for testing the quality of the mean fit", at the end of page 5).

2. `training_length` is implemented in `RNNModel` because by design, `output_chunk_length=1` and the model reuses its output and hidden state to predict additional timestamps. Since users are often interested in predicting several values at once, this value is used to create longer input and train the model in a context closer to what is happening at inference time (see [this other comment](https://github.com/unit8co/darts/issues/1818#issuecomment-1606982414) for more info and the dedicated [doc page](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.rnn_model.html?highlight=rnnmodel#darts.models.forecasting.rnn_model.RNNModel)) and somehow mimic a `output_chunk_length>1`. Because of this, `training_length` cannot be longer than the shortest timeseries used during training/validation.

#635 seems to correspond to the feature you would need in order to use an expanding window, it's in the roadmap but with a low priority. Let us know if you would be interested in implementing it.
